### PR TITLE
Setup PATH using GitHub API

### PR DESCRIPTION
**Pull request by atoomic**

### CODE REVIEW

**Comment by bingos** [comment]
Yeah $GITHUB_PATH is not a Windows environment variable

Needs "Out-File -FilePath $env:GITHUB_PATH -Encoding utf8"

https://github.community/t/migration-from-deprecated-add-path-on-windows/136265

File: .github/workflows/smoke-windows-strawberryperl.yml [hunk 1]
```diff
@@ -26,22 +26,17 @@ jobs:
           perl -MIPC::Cmd -E "say IPC::Cmd::can_run(qq{gcc})"
           perl -MIPC::Cmd -E "say IPC::Cmd::can_run(qq{g++})"
           perl -MIPC::Cmd -E "say IPC::Cmd::can_run(qq{ld})"
-          path C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;%PATH%
+          echo "C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin" ^| Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
```
Comment (bingos) [inline .github/workflows/smoke-windows-strawberryperl.yml:29]: This should be
```
echo "C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;$env:PATH" ^| Out-File -FilePath $env:GITHUB_PATH -Encoding utf8
```
The Strawberry Perl versions of Mingw need to be ahead of the Chocolately versions

**Review by bingos — CHANGES_REQUESTED** [review]

File: .github/workflows/smoke-windows-cygwin.yml [hunk 2]
```diff
@@ -27,23 +27,17 @@ jobs:
         run: |
           dir c:\tools\cygwin
           path
+          echo "c:\tools\cygwin\bin;c:\tools\cygwin\usr\bin" ^| Out-File -FilePath $env:GITHUB_PATH -Encoding utf8
```
Comment (bingos) [inline .github/workflows/smoke-windows-cygwin.yml:30]: Does this need actions/checkout@v2 as well to work?

Comment (atoomic) [reply .github/workflows/smoke-windows-cygwin.yml:30]: going to give it a try

File: .github/workflows/smoke-windows-strawberryperl.yml (hunk 1 above)
Comment (atoomic) [reply .github/workflows/smoke-windows-strawberryperl.yml]: testing with your suggestion via 3aa63b90225c66fec1cddd23b3a108abd99f80a4

File: .github/workflows/smoke-windows-cygwin.yml (hunk 2 above)
Comment (atoomic) [reply .github/workflows/smoke-windows-cygwin.yml:30]: this works, I can see failures from `t/02-xsdynamic.t` which is failing under cygwin & strawberry... 
I skipped it using 5688f4c822a715072f3024531e4e387519392cd2 as this seems similar to `t/03-xsstatic.t`

File: t/02-xsdynamic.t [hunk 3]
```diff
@@ -14,6 +14,7 @@ use Test::More;
 plan skip_all => 'Dynaloading not enabled' if !$Config{usedl} or $Config{usedl} ne 'define';
 plan skip_all => "ExtUtils::CBuilder not installed or couldn't find a compiler"
   unless have_compiler();
+plan skip_all => 'Shared perl library' if $Config{useshrplib} eq 'true';
```
Comment (Leont) [inline t/02-xsdynamic.t:17]: This change looks unrelated to the rest of this change (and I suspect it's wrong)

Comment (bingos) [reply t/02-xsdynamic.t:17]: Yeah, I suspect this is hiding a problem

Comment (atoomic) [reply t/02-xsdynamic.t:17]: I think too that this is hiding another problem

**Comment by atoomic** [comment]
I'm giving up on this one
